### PR TITLE
Fix fuzzing issues

### PR DESCRIPTION
**Pull request by loosebazooka**
- handle invalid json for rekor response
- handle bundle with no tlog entry

### CODE REVIEW

File: sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java [hunk 1]
```diff
@@ -56,7 +57,12 @@ public interface RekorResponse {
   static RekorResponse newRekorResponse(URI entryLocation, String rawResponse)
       throws RekorParseException {
     var type = new TypeToken<Map<String, RekorEntry>>() {}.getType();
-    Map<String, RekorEntry> entryMap = GSON.get().fromJson(rawResponse, type);
+    Map<String, RekorEntry> entryMap;
+    try {
+      entryMap = GSON.get().fromJson(rawResponse, type);
+    } catch (JsonSyntaxException jse) {
+      throw new RekorParseException("Rekor entry json could not be parsed", jse);
```
Comment (vlsi) [inline sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java:64]: WDYT of mentioning the problematic response in the error message?

**Review by vlsi — DISMISSED** [review]

File: sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java (hunk 1 above)
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java:64]: I think the stack trace should show it? but yeah I can just do jse.getMessage() too. I'll add that in.

**Comment by loosebazooka** [comment]
turns out parsing bad data can create npes too.

**Comment by loosebazooka** [comment]
I think we can logger.error the whole rekor response? Is that what you were kinda looking for?

**Review by vlsi — CHANGES_REQUESTED** [review]

File: sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java [hunk 2]
```diff
@@ -53,10 +56,17 @@ public interface RekorResponse {
...
       throws RekorParseException {
     var type = new TypeToken<Map<String, RekorEntry>>() {}.getType();
-    Map<String, RekorEntry> entryMap = GSON.get().fromJson(rawResponse, type);
+    Map<String, RekorEntry> entryMap;
+    try {
+      entryMap = GSON.get().fromJson(rawResponse, type);
+    } catch (JsonSyntaxException | NullPointerException | StringIndexOutOfBoundsException ex) {
+      log.severe("Rekor entry could not be parsed");
+      log.severe(rawResponse);
+      throw new RekorParseException("Rekor entry json could not be parsed: " + ex.getMessage(), ex);
```
[the remark this answers is not in the document]
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java:68]: sgtm... I image that response could be pretty huge, but whatever, this appears to be a pretty edge error case.

File: sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java [hunk 3]
```diff
@@ -53,10 +54,15 @@ public interface RekorResponse {
...
    */
-  static RekorResponse newRekorResponse(URI entryLocation, String rawResponse)
+  public static RekorResponse newRekorResponse(URI entryLocation, String rawResponse)
       throws RekorParseException {
     var type = new TypeToken<Map<String, RekorEntry>>() {}.getType();
-    Map<String, RekorEntry> entryMap = GSON.get().fromJson(rawResponse, type);
+    Map<String, RekorEntry> entryMap;
+    try {
+      entryMap = GSON.get().fromJson(rawResponse, type);
+    } catch (JsonSyntaxException | NullPointerException | StringIndexOutOfBoundsException ex) {
```
Comment (vlsi) [inline sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorResponse.java:63]: we should report npe to gson team

**Review by vlsi — APPROVED** [review]

**Comment by loosebazooka** [comment]
I think fuzzing caught a number format exception issue, but I'll fix that in a followup.

**Pull request merged**